tt_metal/hw/firmware/src/brisc.cc

```diff
@@ -234,19 +238,23 @@ inline void deassert_ncrisc_trisc() {
     uint16_t fw_size16 = mailboxes->launch.ncrisc_kernel_size16;
     ncrisc_kernel_start_offset16 = fw_size16;
 
+#ifdef NCRISC_HAS_IRAM
```

Comment:
redundant to the ifdefs within the routines?  I suggest we just ifdef inside the routines (like you have) and maybe put a comment: // copy to iram for chips that need it

Reply:
ah yeah I needed this because I didn't define `MEM_NCRISC_INIT_IRAM_L1_BASE` in the blackhole dev_mem_map but I can do it and set it to 0?

Reply:
maybe change the fn signature and put the consts inside?

Reply:
eventually maybe we want an arch agnostic base memory map, and then different archs can update relevant addresses / constants